src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -947,7 +983,7 @@ public static Entity getTargetedEntity(final Player player, final double range,
 		final Vector direction = player.getEyeLocation().getDirection().normalize();
 		for (final Entity entity : getEntitiesAroundPoint(origin, range)) {
 			if (entity instanceof Player) {
-				if (((Player) entity).isDead() || ((Player) entity).getGameMode().equals(GameMode.SPECTATOR)) {
+				if (entity.equals(player) || ((Player) entity).isDead() || ((Player) entity).getGameMode().equals(GameMode.SPECTATOR)) {
 					continue;
 				}
 			}

```

Comment:
Can you do type casting? Not sure if the Java version supports it but if it does I would switch

Reply:
this is a java 14 thing

Reply:
> this is a java 14 thing

yep, I just checked, that if we want to support 1.16, we have to use Java 8, otherwise, for 1.17+ we can use Java 16